### PR TITLE
Fixes a bug in the autorelease script and adds options for partial runs of the script.

### PR DESCRIPTION
**Pull request by thisisdhaas**

### CODE REVIEW

File: release.py [hunk 1]
```diff
@@ -20,22 +20,35 @@ def main(args):
...
+        # Commit the update to the repo
+        commit_and_push(fake=args.fake)
 
     # Create a new tag for the release
-    tag_str = tag_release(new_version, fake=args.fake)
+    version_text = 'current' if args.no_commit else 'new'
+    if args.no_tag:
+        print('--no-tag passed, not tagging the {} version.'.format(
+            version_text))
+        tag_str = 'v{}'.format(old_version)
```
Comment (marcua) [inline release.py]: a little nasty :)

File: release.py [hunk 2]
```diff
@@ -20,22 +20,35 @@ def main(args):
...
-    commit_and_push(fake=args.fake)
+        # Commit the update to the repo
+        commit_and_push(fake=args.fake)
 
     # Create a new tag for the release
-    tag_str = tag_release(new_version, fake=args.fake)
+    version_text = 'current' if args.no_commit else 'new'
+    if args.no_tag:
+        print('--no-tag passed, not tagging the {} version.'.format(
+            version_text))
```
Comment (marcua) [inline release.py:38]: spacing?

**Comment by marcua** [comment]
lgtm minus that spacing comment

File: release.py (hunk 2 above)
Comment (thisisdhaas) [inline release.py:38]: we talked offline, it seems okay.

**Comment by marcua** [comment]
Nothing here strikes me as too weird :)

**Pull request merged**